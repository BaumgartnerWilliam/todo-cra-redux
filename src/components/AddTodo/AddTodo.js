import React, { useState } from 'react';
import { Col, Form, Button } from 'react-bootstrap';

const AddTodo = ({ onAddTodo }) => {
  const [inputValue, setInputValue] = useState('');

  const handleSubmit = event => {
    event.preventDefault();
    event.stopPropagation();
    const {
      addTodo: { value }
    } = event.target.elements;

    if (value) {
      // input has some value
      onAddTodo(value);
      setInputValue('');
    }
  };

  const onInputValueChange = ({ target: { value } }) => setInputValue(value);

  return (
    <Col>
      <Form onSubmit={handleSubmit}>
        <Form.Group>
          <Form.Label>Add Todo</Form.Label>
          <Form.Control
            name="addTodo"
            aria-label="add-todo-input"
            placeholder="my todo note"
            onChange={onInputValueChange}
            value={inputValue}
          />
        </Form.Group>
        <div>
          <Button
            aria-label="submit-todo"
            disabled={!inputValue}
            variant="primary"
            type="submit">
            Add
          </Button>
        </div>
      </Form>
    </Col>
  );
};

export default AddTodo;
