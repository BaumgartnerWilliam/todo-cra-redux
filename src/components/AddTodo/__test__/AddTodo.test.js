import React from 'react';
import { render, fireEvent, wait } from '@testing-library/react';
import AddTodo from '../AddTodo';

describe('todo component', () => {
  it('renders without crashing', () => {
    const { container } = render(<AddTodo />);
    expect(container).toBeInTheDocument();
  });

  it('calls the callback with the proper value', async () => {
    const props = {
      onAddTodo: jest.fn()
    };
    const value = 'todo text';
    const { getByLabelText } = render(<AddTodo {...props} />);

    await wait(() => {
      fireEvent.change(getByLabelText('add-todo-input'), {
        target: { value }
      });
    });

    await wait(() => {
      fireEvent.click(getByLabelText('submit-todo'));
    });

    expect(props.onAddTodo).toHaveBeenCalledWith(value);
  });

  it('matches the snapshot', () => {
    const { container } = render(<AddTodo />);
    expect(container).toMatchSnapshot();
  });
});
