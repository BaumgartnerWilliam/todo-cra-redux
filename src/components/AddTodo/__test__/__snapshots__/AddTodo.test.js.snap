// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo component matches the snapshot 1`] = `
<div>
  <div
    class="col"
  >
    <form
      class=""
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          Add Todo
        </label>
        <input
          aria-label="add-todo-input"
          class="form-control"
          name="addTodo"
          placeholder="my todo note"
          value=""
        />
      </div>
      <div>
        <button
          aria-label="submit-todo"
          class="btn btn-primary"
          disabled=""
          type="submit"
        >
          Add
        </button>
      </div>
    </form>
  </div>
</div>
`;
